src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,226 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private User user1;
+    private User user2;
+    private User user3;
+
+    @BeforeAll
+    static void setUpAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        // Technically valid users.
+        // Three of them for testing edge cases, because duplicates are not allowed to store.
+        // Clearing static storage list before each test.
+        user1 = new User();
+        user1.setLogin("uniqueLogin1");
+        user1.setPassword("password");
+        user1.setAge(20);
+
+        user2 = new User();
+        user2.setLogin("uniqueLogin2");
+        user2.setPassword("password");
+        user2.setAge(20);
+
+        user3 = new User();
+        user3.setLogin("uniqueLogin3");
+        user3.setPassword("password");
+        user3.setAge(20);
+
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        user1.setLogin(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user1));
+
+        boolean containsNone = Storage.people.isEmpty();
+        assertTrue(containsNone, "Error. The User was added to the list despite NULL login.");
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        user1.setPassword(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user1));
+
+        boolean containsNone = Storage.people.isEmpty();
+        assertTrue(containsNone, "Error. The User was added to the list despite NULL password.");
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        user1.setAge(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user1));
+
+        boolean containsNone = Storage.people.isEmpty();
+        assertTrue(containsNone, "Error. The User was added to the list despite NULL age.");
+    }
+
+    @Test
+    void register_tooShortLogin_notOk() {
+        user1.setLogin("");
```

Comment:
You don't need 3 separate users to test this feature - according to DRY rule, I think that 2 cases in these tests are redundant. 
Requirement for each parameter is pretty straight-forward: you either have login long enough or too short, so only two cases for testing this behavior are needed.